### PR TITLE
Structured data - Recipes - add description, dates and total time 

### PR DESCRIPTION
**Pull request by mchv**
## What does this change?

Following [initial work on structured data ](https://github.com/guardian/frontend/pull/15556):

* Improve a bit existing code following @TBonnin feedback
* Add `description` field with a fallback until we have `description` field on the model
* Add `totalTime` field
* Fomat correctly time according to google specification

## What is the value of this and can you measure success?

See previous changes.

## Does this affect other platforms - Amp, Apps, etc?

AMP is concerned as well.

### CODE REVIEW

File: common/app/views/fragments/atoms/structureddata/recipe.scala.html [hunk 1]
```diff
@@ -13,26 +15,26 @@
...
-     @recipe.data.time.preparation.map {minutes => "prepTime": "@formatDuration(minutes)",}
-     @recipe.data.time.cooking.map {minutes => "cookTime": "@formatDuration(minutes)",}
-     
-     @* TODO - @recipe.data.time.total.map {minutes => "totalTime": "@formatDuration(minutes)",}  *@
-
-     @recipe.data.serves.map { serves => "recipeYield": "@yieldValue(serves)",}
-
+      @recipe.data.time.preparation.map {minutes => "prepTime": "PT@{minutes}M",}
+      @recipe.data.time.cooking.map {minutes => "cookTime": "PT@{minutes}M",}
+      @recipe.data.time.preparation.combine(recipe.data.time.cooking).map { minutes => "totalTime": "PT@{minutes}M",}
```
Comment (mchv) [inline common/app/views/fragments/atoms/structureddata/recipe.scala.html]: Unfortunately, we can't use `|+|` as `twirl` does not seems to correctly recognize the short operator. As there is only `2` elements to sum it does not hurt the readability.

**Comment by na7hangood** [comment]
can we go with "a tasty recipe with..." for our fullback?

**Comment by mchv** [comment]
@na7hangood sure, done!

File: common/app/views/fragments/atoms/structureddata/recipe.scala.html [hunk 2]
```diff
@@ -90,15 +82,12 @@
...
-            case 0.125 => "1/8"
-            case _ => q 
-        }
+    q match {
+        case qty if qty == qty.toInt => qty.toInt 
+        case 0.75 => "3/4"
+        case 0.5 => "1/2"
+        case 0.25 => "1/4"
+        case 0.125 => "1/8"
+        case _ => q
```
Comment (TBonnin) [inline common/app/views/fragments/atoms/structureddata/recipe.scala.html]: Can you make sure this block always returns a `String`? 😉

File: common/app/views/fragments/atoms/structureddata/recipe.scala.html [hunk 3]
```diff
@@ -1,6 +1,8 @@
 @(recipe: model.content.RecipeAtom)(implicit request: RequestHeader)
 @import play.api.libs.json.Json
 @import views.support.{ImgSrc, GoogleStructuredData}
+@import org.joda.time.DateTime
+@import cats.implicits._
```
Comment (TBonnin) [inline common/app/views/fragments/atoms/structureddata/recipe.scala.html]: We don't use `cats` or any FP library in `frontend` anymore. We had a long discussion about it. If you want to introduce `cats` please come talk to us (@guardian/dotcom-platform) before doing so.

File: common/app/views/fragments/atoms/structureddata/recipe.scala.html (hunk 1 above)
Comment (TBonnin) [inline common/app/views/fragments/atoms/structureddata/recipe.scala.html]: See my comments above about `cats`

**Review by TBonnin — CHANGES_REQUESTED** [review]

File: common/app/views/fragments/atoms/structureddata/recipe.scala.html (hunk 3 above)
Comment (mchv) [inline common/app/views/fragments/atoms/structureddata/recipe.scala.html]: @TBonnin sure I will as it make me a bit sad :(

File: common/app/views/fragments/atoms/structureddata/recipe.scala.html (hunk 2 above)
Comment (mchv) [inline common/app/views/fragments/atoms/structureddata/recipe.scala.html]: yep

**Comment by mchv** [comment]
@TBonnin have updated my changes

**Review by TBonnin — APPROVED** [review]

**Pull request merged**